torch/_higher_order_ops/auto_functionalize.py

```diff
@@ -820,13 +909,76 @@ def auto_functionalized_v2_fake(
 @auto_functionalized_v2.py_impl(ProxyTorchDispatchMode)
 def auto_functionalized_v2_proxy(
     mode,
-    _mutable_op: OpOverload,
-    **kwargs: dict[str, Any],
+    _mutable_op: _MutableOpType,
+    **kwargs: Any,
 ) -> tuple[Any, tuple[Tensor, ...]]:
+    if isinstance(_mutable_op, HigherOrderOperator):
+
```

Comment:
Can you describe what is happening here and why? My understanding is that we're in a siutation like `auto_functionalized(invoke_subgraph, f, x)` and the way we are resolving this is that we're going to trace invoke_subgraph(f, x), find the node, and the re-use the graph created there

Reply:
yeah, you're right. I breifly explain why we need to materialize but didn't explain why the implementation is like this. I'll add it to the note